[charts/gateway]Us896244 dbupgrade job

uppoju opened the pull request:
**Description of the change**

Added DB upgrade job for creating OTK DB
Supported Demo DB for OTK
Add read only connection support

**Benefits**
Customers can use OTK without configuring external DB for validation purpose

**Drawbacks**

NA

**Applicable issues**

NA
  - fixes #

**Additional information**


**Checklist** <!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[charts/gateway]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

Commit:
US896244: DB upgrade job

Commit:
Resolve merge conflicts

Commit:
Fix read only connection check

Commit:
Update readme with new parameters

Commit:
modify production values

Gazza7205 commented on charts/gateway/production-values.yaml:
```diff
@@ -528,10 +535,22 @@ otk:
   # Install OTK Health check bundle on gateway. Uses config map.
   # Alternatively the bundle can be loaded using config map external to helm (useExisting: true)
-  # The bundle content can be found at https://github.com/CAAPIM/apim-charts/blob/stable/charts/gateway/bundles/otk-healthcheck.bundle
+  # The bundle content can be found at https://github.com/CAAPIM/apim-charts/blob/stable/charts/gateway/bundles/otkhealthcheck.bundle
```
otkhealthcheck.bundle doesn't appear to exist

uppoju replied:
Done

Gazza7205 commented on charts/gateway/values.yaml:
```diff
@@ -478,7 +478,9 @@ service:
 # Note: In dual gateway installation, restart the pods after OTK install or upgrade.
 otk:
-  enabled: false
+  enabled: true
```
should be disabled (false) by default

uppoju replied:
Done

Gazza7205 reviewed (changes requested):
There's a ref in production-values.yaml that needs to be updated and default for otk.enabled should be false in values.yaml.

Chart version should be 3.0.13

Please check the linting errors with [chart-tester](https://github.com/helm/chart-testing/tree/main) locally - if the check fails the release pipeline will too

Commit:
Address review comments

Commit:
Change chart version

uppoju committed:
Merge branch 'stable' into US896244_dbupgrade_job

Commit:
Fixed lint error

Commit:
Merge branch 'US896244_dbupgrade_job' of https://github.com/uppoju/apim-charts into US896244_dbupgrade_job

Commit:
Update production values as per the values yaml file

uppoju commented:
> There's a ref in production-values.yaml that needs to be updated and default for otk.enabled should be false in values.yaml.
> 
> Chart version should be 3.0.13
> 
> Please check the linting errors with [chart-tester](https://github.com/helm/chart-testing/tree/main) locally - if the check fails the release pipeline will too

Addressed all the comments

Gazza7205 reviewed (approved):
LGTM

Commit:
Add ddl user name and password to values and production values files
